src/decompress.c

```diff
@@ -115,8 +129,17 @@ zseek_reader_t *zseek_reader_open_full(zseek_read_file_t user_file,
     }
     reader->cache = cache;
 
+    zseek_buffer_t *cbuf = zseek_buffer_new(0);
+    if (!cbuf) {
+        set_error(errbuf, "buffer creation failed");
+        goto fail_w_cache;
+    }
+    reader->cbuf = cbuf;
+
     return reader;
 
+fail_w_cache:
```

Comment:
Note: there is a much more robust pattern for this. Make *free() silently return on NULL inputs and use a single label for error handling. Not strictly necessary for this PR, but something to keep in mind for future refactoring.

Reply:
Indeed, this is a habit I have picked up, which starts to prove hard to maintain. Will consider this refactor for sure, thank you!